core/unit_test/CMakeLists.txt

```diff
@@ -65,8 +65,11 @@ KOKKOS_INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR})
 KOKKOS_INCLUDE_DIRECTORIES(REQUIRED_DURING_INSTALLATION_TESTING ${CMAKE_CURRENT_SOURCE_DIR})
 KOKKOS_INCLUDE_DIRECTORIES(${KOKKOS_SOURCE_DIR}/core/unit_test/category_files)
 
-add_executable(TestCompileOnly TestDetectionIdiom.cpp)
-target_link_libraries(TestCompileOnly PRIVATE Kokkos::kokkos)
+KOKKOS_ADD_EXECUTABLE_AND_TEST(
+  TestCompileOnly
+  SOURCES
+  TestDetectionIdiom.cpp
+)
```

Comment:
Well I only meant to add an executable.  Does `KOKKOS_ADD_EXECUTABLE` work?

Reply:
Yes, I amended the PR with the change